src/app/core/pricing/pricing-tier.service.ts

```diff
@@ -23,7 +23,7 @@ export class PricingTierService {
         this.isVolumeDiscountActive$.next(value ?? !this.isVolumeDiscountActive$.value);
     }
 
-    updateDiscount(itemCount: number): void {
-        this.toggleDiscount(itemCount > ITEM_COUNT_THRESHOLD);
+    updateDiscount(cartWeight: number): void {
```

Comment:
This is good. Maybe one thing I did not specify, we only programmatically toggle the discount when going from 30% to 45%, but no the other way around.

It may seem counterintuituive, but the logic is that weight may go down temporarily when people decide what to order and end up adding, removing, adding, etc. So we want to cover the case where people order above the threshold when they thought they would order for 30%, but give them the benefit of the doubt that when the toggle gets activated, it should not get deactivated. (this will of course get further managed on our side when creating invoices)